compiler/rustc_metadata/src/creader.rs

```diff
@@ -598,7 +598,10 @@ impl<'a> CrateLoader<'a> {
         // don't want to match a host crate against an equivalent target one
         // already loaded.
         let root = library.metadata.get_root();
-        Ok(Some(if locator.triple == self.sess.opts.target_triple {
+        // Proc-macro crates are always compiled for the host target, so they should be reused even if we're cross-compiling.
```

Comment:
How does this interact with `-Zdual-proc-macro`? It is used by bootstrap to ensure that cross-compilation of rustc works fine. It causes both the host and target version of a proc macro to be registered in the crate metadata allowing either to be used at runtime depending on if rustc runs on the original host or the target.

Reply:
I'm not sure. How can I test?

Reply:
The for sure test would be to try cross-compiling rustc and then try to compile a program linking against `rustc_driver` using the cross-compiled rustc. Alternatively I think the following would also work:

* compile my_proc_macro for host in dir a
* compile my_proc_macro for target with the same filename in dir b
* compile my_crate for target with the host the host my_proc_macro as `--extern` and both `a` and `b` as `-L` and `-Zdual-proc-macros` as argument. (make sure to `use my_proc_macro;` otherwise rustc won't actually load it)
* try to link against my_crate using a compiler for the target, but the same version of rustc.

Reply:
I tried unsuccessfully to test this :( https://zulip-archive.rust-lang.org/131828tcompiler/56888HowdoIcrosscompilerustcforadifferenthost.html